### PR TITLE
Draft intro page

### PR DESCRIPTION
**Pull request by fonnesbeck**

### CODE REVIEW

File: docs/source/intro.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+Introduction
+************
+
+:Date: 2017-07-06
+:Authors: Chris Fonnesbeck
+:Contact: fonnesbeck@gmail.com
+:Web site: http://github.com/pymc-devs/pymc3
+
+
+Purpose
```
Comment (aseyboldt) [inline docs/source/intro.rst]: I don't think we need a section title. It's kind of obvious that we introduce pymc here :-)

File: docs/source/intro.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+=======
+
+PyMC3 is a probabilistic programming module for Python that allows users to fit Bayesian models using a variety of numerical methods, most notably, Markov chain Monte Carlo (MCMC) and variational inference (VI). Its flexibility and extensibility make it applicable to a large suite of problems. Along with core model specification and fitting functionality, PyMC3 includes methods for summarizing output, plotting, goodness-of-fit and model diagnostics.
```
Comment (aseyboldt) [inline docs/source/intro.rst]: The rules for commas in English are a bit of a mystery to me, but do we need the comma after "most notably"?
"suite of problems" sounds a bit strange to me (but there are a lot of google hits, so maybe that is a thing after all). How about something like "It it being used in biology, astronomy, physics, the social sciences, business and finance and can be extended to fit new needs" (we might want to check that list however). I guess that would cover "flexibility".
Maybe "... fitting functionality, PyMC3 can visualize its results and provides model diagnostics."?

File: docs/source/intro.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+
+
+Features
+========
+
+PyMC3 strives to make Bayesian modeling as simple and painless as possible, thereby allowing users to focus on their scientific problem, rather than on the methods used to solve it. Here is a partial list of its features:
```
Comment (aseyboldt) [inline docs/source/intro.rst]: "thereby allowing" -> "to allow"?
I don't think we need the second sentence. Clear from context.

File: docs/source/intro.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+* Fits Bayesian statistical models using modern techniques, including MCMC and VI.
```
Comment (aseyboldt) [inline docs/source/intro.rst]: Third person sounds a bit strange. Maybe imperative? Or just "Modern techniques for fitting Bayesian stats models..."
Replace MCMC by hamiltonian MCMC and no abbreviations.

File: docs/source/intro.rst [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+* Includes a large suite of well-documented statistical distributions.
+
+* Uses Theano as the computational backend, allowing for fast expression evaluation, automatic gradient calculation, and GPU computing.
+
+* Includes a module for Gaussian process modeling.
```
Comment (aseyboldt) [inline docs/source/intro.rst]: "Build-in support for Gaussian process modeling"

File: docs/source/intro.rst [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+* Creates summaries including tables and plots.
```
Comment (aseyboldt) [inline docs/source/intro.rst]: Kind of obvious, I don't think we need to mention that one.

File: docs/source/intro.rst [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+* Several convergence diagnostics are available.
```
Comment (aseyboldt) [inline docs/source/intro.rst]: "Model checking and convergence detection"

File: docs/source/intro.rst [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+* Extensible: easily incorporates custom step methods and unusual probability
+  distributions.
+
+* Bayesian models can be embedded in larger programs, and results can be analyzed
+  with the full power of Python.
+
+
+What's new in version 3
```
Comment (aseyboldt) [inline docs/source/intro.rst:39]: Mabe just mention that it is a rewrite, but not much detail. That could be on a separate page pymc2 vs pymc3. I could be very wrong about this, but so far I didn't have the impression that most pymc3 users also knew pymc2. For those users the comparison is kind of pointless. Also, this kind of ends up repeating the feature list from above.
The important part of this is I think mentioned in the history section.

File: docs/source/intro.rst [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+* Much more!
+
+While the addition of Theano adds a level of complexity to the development of PyMC, fundamentally altering how the underlying computation is performed, we have worked hard to maintain the elegant simplicity of the original PyMC model specification syntax. 
+
+Usage Overview
+==============
+
+First, import the PyMC3 functions and classes you will need for building your model. You can import the entire module via `import pymc3 as pm`, or just bring in what you need::
+
+    from pymc3 import Model, Normal, invlogit, Binomial, sample, traceplot
```
Comment (aseyboldt) [inline docs/source/intro.rst]: Pretty much everywhere we are using `import pymc3 as pm`. So I think we should do the same thing here.

File: docs/source/intro.rst [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+
+        # Prior distributions for latent variables
+        alpha = Normal('alpha', 0, sd=100)
+        beta = Normal('beta', 0, sd=100)
+
+        # Linear combinations of parameters
+        theta = invlogit(alpha + beta*dose)
+
+        # Model likelihood
+        deaths = Binomial('deaths', n=n, p=theta, observed=np.array([0, 1, 3, 5]))
```
Comment (aseyboldt) [inline docs/source/intro.rst]: Move the definition of the observations before the model def?

File: docs/source/intro.rst [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+
+Save this file, then from a python shell (or another file in the same directory), call::
+
+	with bioassay_model:
```
Comment (aseyboldt) [inline docs/source/intro.rst]: The indentation is strange

**Review by aseyboldt — COMMENTED** [review]
I wrote down some random stuff, feel free to ignore if you don't agree (english isn't even my native language after all ;)

File: docs/source/intro.rst (hunk 11 above)
Comment (fonnesbeck) [reply docs/source/intro.rst]: I don't se a problem locally. I will double check.

File: docs/source/intro.rst (hunk 9 above)
Comment (fonnesbeck) [reply docs/source/intro.rst]: Both are valid. I've noted the option for `pm` above.

**Comment by fonnesbeck** [comment]
Thanks! I've integrated several of these suggestions.

File: docs/source/intro.rst (hunk 8 above)
Comment (fonnesbeck) [reply docs/source/intro.rst:39]: I've tried to restrict this list to differences. We do get a fair number of users moving over, and probably will get more now that v.3 is released.

**Comment by aseyboldt** [comment]
@fonnesbeck Is this finished?

**Comment by fonnesbeck** [comment]
Yeah, I think so. It can be improved with subsequent PRs if desired.

**Pull request merged**